pkg/provider/gke/gke.go

```diff
@@ -60,12 +64,19 @@ type GKE struct {
 // NewGKEClient sets the GKE client used when performing GKE requests.
 func (c *GKE) NewGKEClient(*kingpin.ParseContext) error {
 	// Set the auth env variable needed to the gke client.
-	if c.AuthFile != "" {
-		os.Setenv("GOOGLE_APPLICATION_CREDENTIALS", c.AuthFile)
-	} else if c.AuthFile = os.Getenv("GOOGLE_APPLICATION_CREDENTIALS"); c.AuthFile == "" {
-		log.Fatal("GOOGLE_APPLICATION_CREDENTIALS env is empty. Please run with -a key.json or run `export GOOGLE_APPLICATION_CREDENTIALS=key.json`")
+	if c.Auth != "" {
+	} else if c.Auth = os.Getenv("GOOGLE_APPLICATION_CREDENTIALS"); c.Auth == "" {
+		log.Fatal("no auth provided! Need to either set the auth flag or the GOOGLE_APPLICATION_CREDENTIALS env variable")
 	}
-	cl, err := gke.NewClusterManagerClient(context.Background())
+
+	// When the auth variable points to a file
+	// put the file content in the variable.
+	if content, err := ioutil.ReadFile(c.Auth); err == nil {
+		c.Auth = string(content)
+	}
+	opts := option.WithCredentialsJSON([]byte(c.Auth))
```

Comment:
```suggestion
        if decoded, err := base64.StdEncoding.DecodeString(c.Auth); err == nil {
		c.Auth = string(decoded)
	}
	opts := option.WithCredentialsJSON([]byte(c.Auth))
```

Reply:
We need to check if the string is base64 encoded and decode it if necesary.

Reply:
Actually I used the answer from https://stackoverflow.com/questions/8571501/how-to-check-whether-a-string-is-base64-encoded-or-not/20145780

to determine when the string is base64 encoded. This way the error handling is more deterministic.

Please test and suggest changes if needed.

Reply:
I have tested it with Github Actions also and its working.